packages/block-editor/src/components/block-switcher/style.scss

```diff
@@ -88,26 +88,41 @@
 		@include formatting-button-style__focus();
 	}
 }
-
+// we double the max-width for it to fit both the preview & content
+// but we keep the min width the same for the border to work.
 .components-popover:not(.is-mobile).block-editor-block-switcher__popover .components-popover__content {
 	min-width: 300px;
-	max-width: 340px;
+	max-width: calc(340px * 2);
+	max-height: 500px;
+	display: flex;
+	// we need to differ those styles to popover__container so we reset them here
+	box-shadow: none;
+	background: none;
 }
 
 .block-editor-block-switcher__popover .components-popover__content {
 	@include break-medium {
 		position: relative;
 
+		.components-popover__container {
+			box-shadow: $shadow-popover;
+			background: $white;
+			min-width: 300px;
+			max-width: 340px;
+			width: 50%;
+		}
+
 		.block-editor-block-preview {
-			border: $border-width solid $light-gray-500;
+			border-left: $border-width solid $light-gray-500;
 			box-shadow: $shadow-popover;
 			background: $white;
-			position: absolute;
-			left: 100%;
-			top: -1px;
-			bottom: -1px;
 			width: 300px;
 			height: auto;
+			// sticky helps us keep the preview at the top when scrolling
+			position: sticky;
```

Comment:
If the parent is a flex container, can't we use `align-items` instead?

Reply:
the problem is not much about having the element at the top, but at keeping it visible while scrolling, imagine having a long list of styles with a small height, the user might scroll down to see a style, making the preview invisible
this is what will happen if we don't have sticky
![image](https://user-images.githubusercontent.com/6165348/62165716-fc984200-b316-11e9-900c-384cd642b980.png)